Scripts/gamedata.gd

```diff
@@ -920,43 +919,87 @@ func on_item_changed(newdata: Dictionary, olddata: Dictionary):
-			changes_made = remove_reference(Gamedata.data.items, "core", "items", \
-			res_id, item_id) or changes_made
+			changes_made = remove_reference(Gamedata.data.items, "core", "items", res_id, item_id) or changes_made
 
 	# Add references for new resources, nothing happens if they are already present
 	for res_id in new_resource_ids:
-		changes_made = add_reference(Gamedata.data.items, "core", "items", \
-		res_id, item_id) or changes_made
+		changes_made = add_reference(Gamedata.data.items, "core", "items", res_id, item_id) or changes_made
+
+	# Collect unique skill IDs from old and new recipes
+	var old_skill_ids: Dictionary = {}
+	var new_skill_ids: Dictionary = {}
+
+	# Collect skill IDs from old recipes
+	for recipe in olddata.get("Craft", []):
+		if recipe.has("skill_requirement"):
+			var old_skill_req_id = recipe["skill_requirement"].get("id", "")
+			if old_skill_req_id != "":
+				old_skill_ids[old_skill_req_id] = true
+		if recipe.has("skill_progression"):
+			var old_skill_prog_id = recipe["skill_progression"].get("id", "")
+			if old_skill_prog_id != "":
+				old_skill_ids[old_skill_prog_id] = true
+
+	# Collect skill IDs from new recipes
+	for recipe in newdata.get("Craft", []):
+		if recipe.has("skill_requirement"):
+			var new_skill_req_id = recipe["skill_requirement"].get("id", "")
+			if new_skill_req_id != "":
+				new_skill_ids[new_skill_req_id] = true
+		if recipe.has("skill_progression"):
+			var new_skill_prog_id = recipe["skill_progression"].get("id", "")
+			if new_skill_prog_id != "":
+				new_skill_ids[new_skill_prog_id] = true
+
+	# Check for "Ranged" property in olddata and newdata, and collect skill IDs
+	if olddata.has("Ranged") and olddata["Ranged"].has("used_skill"):
+		var old_skill_id = olddata["Ranged"]["used_skill"].get("skill_id", "")
+		if old_skill_id != "":
+			old_skill_ids[old_skill_id] = true
+
+	if newdata.has("Ranged") and newdata["Ranged"].has("used_skill"):
+		var new_skill_id = newdata["Ranged"]["used_skill"].get("skill_id", "")
+		if new_skill_id != "":
+			new_skill_ids[new_skill_id] = true
+
+	# Remove old skill references that are not in the new list
+	for old_skill_id in old_skill_ids.keys():
+		if not new_skill_ids.has(old_skill_id):
+			changes_made = remove_reference(Gamedata.data.skills, "core", "items", old_skill_id, item_id) or changes_made
+
+	# Add new skill references
+	for new_skill_id in new_skill_ids.keys():
+		changes_made = add_reference(Gamedata.data.skills, "core", "items", new_skill_id, item_id) or changes_made
 
 	# Save changes if any modifications were made
 	if changes_made:
 		save_data_to_file(Gamedata.data.items)
 		save_data_to_file(Gamedata.data.wearableslots)
+		save_data_to_file(Gamedata.data.skills)
 		print_debug("Item changes saved successfully.")
 	else:
 		print_debug("No changes were made to item.")
-
-
 # An item is being deleted from the data
 # We have to remove it from everything that references it
 func on_item_deleted(item_id: String):
```

Comment:
NIT: Function too long and indentation is too deep, consider splitting the body into multiple smaller functions and use early returns/continue to reduce indentation levels.

I am leaving it up to you - this is nothing critical, not a mandatory change.

Reply:
I added a update_item_skill_references function to break it up